File: 351003/Pulish/publisher/app/core/error_handler.py
from fastapi.responses import JSONResponse
from fastapi import FastAPI
from app.core.exceptions import AppException


def register_exception_handlers(app: FastAPI):
    @app.exception_handler(AppException)
    def handle_app_exception(_, exc: AppException):
        return JSONResponse(
            status_code=exc.status,
            content={
                "errorMessage": exc.message,
                "errorCode": exc.code
            }
        )

File: 351004/Leshok/lab3/publisher/src/utils/errorHandler.js
function errorHandler(err, req, res, next) {
  const status = err.status || 500;
  const code = err.code || (status === 500 ? '50000' : `${status}00`);
  res.status(status).json({ errorMessage: err.message, errorCode: code });
}

module.exports = errorHandler;